isic_archive/models/dataset.py

```diff
@@ -580,14 +581,16 @@ def applyMetadata(self, dataset, metadataFile, save):
         # Avoid circular import
         from .image import Image
 
-        metadataFileStream = File().download(metadataFile, headers=False)()
+        metadataFileStream = io.BytesIO()
+        for chunk in File().download(metadataFile, headers=False)():
+            metadataFileStream.write(chunk)
 
         images = []
         metadataErrors = []
         metadataWarnings = set()
 
         try:
-            csvReader = csv.DictReader(generateLines(metadataFileStream))
+            csvReader = csv.DictReader(metadataFileStream.getvalue().decode('utf-8').splitlines())
```

Comment:
I think we should use `'utf-8-sig'` here. If the file includes a BOM at the beginning, it will properly decode it; see: https://docs.python.org/3/library/codecs.html#encodings-and-unicode If the file doesn't include a BOM, it should behave the same as `'utf-8'`: https://stackoverflow.com/a/44573867

This is something we could easily write a test for, by generating a sample CSV file with a BOM and without.